lua/astrocommunity/utility/toggleterm-manager/init.lua

```diff
@@ -0,0 +1,50 @@
+return {
+  {
+    "ryanmsnyder/toggleterm-manager.nvim",
+    opts = function(_, opts)
+      local toggleterm_manager = require "toggleterm-manager"
+      local actions = toggleterm_manager.actions
+
+      local mappings = {
+        i = {
+          ["<CR>"] = { action = actions.toggle_term, exit_on_action = true }, -- toggles terminal open/closed
+          ["<C-i>"] = { action = actions.create_term, exit_on_action = false }, -- creates a new terminal buffer
+          ["<C-d>"] = { action = actions.delete_term, exit_on_action = false }, -- deletes a terminal buffer
+          ["<C-r>"] = { action = actions.rename_term, exit_on_action = false }, -- provides a prompt to rename a terminal
+        },
+        n = {
+          ["<CR>"] = { action = actions.toggle_term, exit_on_action = true }, -- toggles terminal open/closed
+          ["r"] = { action = actions.rename_term, exit_on_action = false }, -- provides a prompt to rename a terminal
+          ["d"] = { action = actions.delete_term, exit_on_action = false }, -- deletes a terminal buffer
+          ["n"] = { action = actions.create_term, exit_on_action = false }, -- creates a new terminal buffer
+        },
+      }
+
+      if not opts.mappings then
+        opts.mappings = mappings
+      else
+        opts.mappings = vim.tbl_deep_extend("force", opts.mappings, mappings)
+      end
+    end,
+
+    config = function(_, opts)
+      local toggleterm_manager = require "toggleterm-manager"
+      toggleterm_manager.setup(opts)
+    end,
+    dependencies = {
+      "akinsho/nvim-toggleterm.lua",
+      "nvim-telescope/telescope.nvim",
+      "nvim-lua/plenary.nvim", -- only needed because it's a dependency of telescope
```

Comment:
```suggestion
```

This isn't necessary because because the real installation of `telescope` should handle it's own dependencies

Reply:
It was treated as a dependency of the toggleterm-manager, will it be automatically handled as well?

Reply:
Ah I see, I misinterpreted your comment as meaning it's a soft dependency. But it is actually a hard dependency used in the source code. Better to leave it but remove the unnecessary and confusing comment